chainerx_cc/chainerx/routines/creation.h

```diff
@@ -120,4 +122,6 @@ Array Linspace(
         const nonstd::optional<Dtype>& dtype = nonstd::nullopt,
         Device& device = GetDefaultDevice());
 
+std::vector<Array> Meshgrid(const std::vector<Array>& arrays, const nonstd::optional<std::string>& indexing);
```

Comment:
How about defining an enum for the indexing argument similar to https://github.com/chainer/chainer/blob/master/chainerx_cc/chainerx/routines/pooling.h#L18-L21 and take an instance of that as the argument instead? The input validation can be done in the Python binding glue code and would not be needed here.

Reply:
Input validation moved to binding code.
Have directly used string instead as we are certain that there will only be 2 modes, I felt having a `enum` for these two would be over-kill. Your thoughts?

Reply:
I see your point but I must say I think it's a better API design choice reducing possible errors. It also simplifies input validation.

Reply:
Have used `enum`, have to agree it is more clear in terms of valid choices. Thank You.